contracts/SpokePool.sol

```diff
@@ -1233,28 +1294,11 @@ abstract contract SpokePool is
 
         // If leaf's amountToReturn is positive, then send L2 --> L1 message to bridge tokens back via
         // chain-specific bridging method.
-        if (relayerRefundLeaf.amountToReturn > 0) {
-            _bridgeTokensToHubPool(relayerRefundLeaf);
-
-            emit TokensBridged(
-                relayerRefundLeaf.amountToReturn,
-                relayerRefundLeaf.chainId,
-                relayerRefundLeaf.leafId,
-                relayerRefundLeaf.l2TokenAddress,
-                msg.sender
-            );
-        }
+        if (amountToReturn > 0) {
+            _bridgeTokensToHubPool(amountToReturn, l2TokenAddress);
 
-        emit ExecutedRelayerRefundRoot(
-            relayerRefundLeaf.amountToReturn,
-            relayerRefundLeaf.chainId,
-            relayerRefundLeaf.refundAmounts,
-            rootBundleId,
-            relayerRefundLeaf.leafId,
-            relayerRefundLeaf.l2TokenAddress,
-            relayerRefundLeaf.refundAddresses,
-            msg.sender
-        );
+            emit TokensBridged(amountToReturn, chainId(), leafId, l2TokenAddress);
```

Comment:
We switch here from `relayerRefundLeaf.chainId` to `chainId()`. Is that just to reduce the number of arguments that need to be supplied to this function?

Reply:
hmm no I think its either an oversight or perhaps a gas optimization. Let me play with it and see whether the MLOAD or the block.chainId emits more gas

Reply:
Turns out passing in _chainId is quite a bit cheaper, perhaps the internal function call is expensive